core/src/main/java/org/apache/carbondata/core/index/TableIndex.java

```diff
@@ -595,6 +595,29 @@ public void onEvent(Event event, OperationContext opContext) {
       throws IOException {
     List<Segment> segments = getCarbonSegments(allsegments);
     Map<String, Long> blockletToRowCountMap = new HashMap<>();
+    for (Segment segment : segments) {
+      List<CoarseGrainIndex> indexes = defaultIndex.getIndexFactory().getIndexes(segment);
+      for (CoarseGrainIndex index : indexes) {
+        if (null != partitions) {
+          // if it has partitioned index but there is no partitioned information stored, it means
+          // partitions are dropped so return empty list.
+          if (!index.validatePartitionInfo(partitions)) {
+            return new HashMap<>();
+          }
+        }
+        index.getRowCountForEachBlock(segment, partitions, blockletToRowCountMap);
+      }
+    }
+    return blockletToRowCountMap;
+  }
+
+  /**
+   * Prune the index of the given segments and return the Map of blocklet path and row count
+   */
+  public Map<String, Long> getRowCountForBlock(List<Segment> allsegments,
```

Comment:
Please give a more proper name and modify comment

Reply:
done